README.md

```diff
@@ -56,39 +56,12 @@ func main() {
 }
 ```
 
-## Table of Contents
```

Comment:
For a long readme like this, i think its important to have a table of contents

Reply:
As explained in the commit message, GitHub auto-generates a table of contents for every Markdown document 
<img width="310" alt="Screen Shot 2021-09-16 at 20 25 01" src="https://user-images.githubusercontent.com/887/133665167-c8887058-23f1-4c6d-9694-e9de57792728.png">

Reply:
whoa! that's fancy. Sounds good to me!